File: java/src/test/java/io/binac/leetcode/CountAndSayTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CountAndSayTest {
    private static final CountAndSay.Solution1 SOLUTION1 = new CountAndSay.Solution1();

    private final int N1 = 1;
    private final String EXPECTED1 = "1";

    private final int N2 = 4;
    private final String EXPECTED2 = "1211";

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.countAndSay(N1));
        assertEquals(EXPECTED2, SOLUTION1.countAndSay(N2));
    }
}

File: cpp/unittests/CountAndSayTest.cpp
#include "CountAndSay.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(CountAndSay, Solution38_1) {
  const int N1 = 1;
  const std::string Expected1 = "1";
  EXPECT_EQ(Expected1, Solution38_1().countAndSay(N1));

  const int N2 = 4;
  const std::string Expected2 = "1211";
  EXPECT_EQ(Expected2, Solution38_1().countAndSay(N2));
}